packages/api/src/rich-text/detection.ts

```diff
@@ -41,7 +41,7 @@ export function detectFacets(text: UnicodeString): Facet[] | undefined {
     // links
     const re = URL_REGEX
     while ((match = re.exec(text.utf16))) {
-      let uri = match[2]
+      let uri = match[2].toLowerCase()
```

Comment:
I think it will introduce another problem because URI is now manipulated, and it's a return value of this function.

Reply:
Oh good point ! I think we can create a separate variable to check if the rawUri is a valid url. If yes, we return the rawUri instead.